staking/types/validator.go

```diff
@@ -297,6 +284,19 @@ func UpdateValidatorFromEditMsg(validator *Validator, edit *EditValidator) error
 			validator.SlotPubKeys = append(validator.SlotPubKeys[:index], validator.SlotPubKeys[index+1:]...)
 		}
 	}
+
```

Comment:
Changed the order to put add after remove. To avoid the problem when add/remove the same existing key.